scripts/asr_language_modeling/ngram_lm/opengrm.py

```diff
@@ -0,0 +1,285 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Utility methods to be used to merge arpa N-gram language models (LMs), 
+culculate perplexity of resulted LM, and make binary KenLM from it.
+
+Minimun usage example to merge two N-gram language models with weights:
+alpha * ngram_a + beta * ngram_b = 2 * ngram_a + 1 * ngram_b
+
+python3 opengrm.py  --kenlm_bin_path /workspace/nemo/decoders/kenlm/build/bin/build_binary \
+                    --arpa_a /path/ngram_a.kenlm.tmp.arpa \
+                    --alpha 2 \
+                    --arpa_b /path/ngram_b.kenlm.tmp.arpa \
+                    --beta 1 \
+                    --out_path /path/out \
+
+
+Merge two N-gram language models and calculate its perplexity with test_file.
+python3 opengrm.py  --kenlm_bin_path /workspace/nemo/decoders/kenlm/build/bin/build_binary \
+                    --arpa_a /path/ngram_a.kenlm.tmp.arpa \
+                    --alpha 0.5 \
+                    --arpa_b /path/ngram_b.kenlm.tmp.arpa \
+                    --beta 0.5 \
+                    --out_path /path/out \
+                    --tokenizer_model_file /path/to/model_tokenizer.nemo \
+                    --test_file /path/to/test_manifest.json \
+                    --force
+"""
+
+import argparse
+import os
+import subprocess
+import sys
+
+import torch
+
+import nemo.collections.asr as nemo_asr
+from nemo.collections.asr.parts.submodules.ctc_beam_decoding import DEFAULT_TOKEN_OFFSET
+from nemo.collections.common.tokenizers.sentencepiece_tokenizer import SentencePieceTokenizer
+from nemo.utils import logging
+
+
+def ngrammerge(arpa_a, alpha, arpa_b, beta, arpa_c, force):
```

Comment:
Docstrings in functions or inline would be nice for long term maintenance

Reply:
added